src/config_libconfig.c

```diff
@@ -37,6 +36,48 @@ static inline int lcfg_lookup_bool(const config_t *config, const char *path, boo
 	return ret;
 }
 
+const char *xdgConfigHome(void) {
+	char *xdgh = getenv("XDG_CONFIG_HOME");
+	char *home = getenv("HOME");
+
+	if (!xdgh) {
+		if (!home) return NULL;
+
+		xdgh = malloc(strlen(home) + 8 + 1);
+
+		if (!xdgh) return NULL;
+
+		strcpy(xdgh, home);
+		strcat(xdgh, "/.config");
+	}
+
+	return xdgh;
+}
+
+const char * const * xdgConfigDirectories(void) {
+	char *dirs, *path, *xdgd = getenv("XDG_CONFIG_DIRS");
+	char **dir_list = {0};
+	unsigned int sep = 0;
+
+	if (!xdgd) xdgd = ":/etc/xdg:";
+
+	dirs = malloc(strlen(xdgd) + 1);
+	strcpy(dirs, xdgd);
```

Comment:
`strdup` is more preferable. Also don't forget to check for allocation failures.

```suggestion
	dirs = strdup(xdgd);
	CHECK(dirs != NULL);
```